### ISSUE TITLE
CS8604 showing regardless of assert on the same property.

### ISSUE THREAD

**Issue by Tape-Worm** [dotnet/roslyn#84718]
I'm using an array: `Plane[] planes`.  This array is populated in the constructor of my class.

I'm seeing a CS8604 regarding using of an element in this array in a method, even though I'm doing an assert on the property.

```csharp
Debug.Assert(_planes[0].VertexBufferView is not null, "Vertex buffer not available for plane 0.");
Debug.Assert(_planes[0].IndexBuffer is not null, "Index buffer not available for plane 0.");

builder.AssignTextureView(_texture, ShaderStage.Pixel, TextureUsage.ReadOnly)
       .AssignBufferView(_planes[0].VertexBufferView, ShaderStage.Vertex, BufferUsage.VertexBuffer) // <-- this line, first parameter.
       .Build(_planes[0].IndexBuffer, _sphere.IndexCount, _pipelineState), // <-- And this line, first parameter.
```

There is no code after the `Assert` that would modify the array to potentially make the element or its property null, why is this warning popping up?

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/48945

**Comment by CyrusNajmabadi** [comment]
Roslyn does not track nullability of elements within an array. You can pull this data out into a local and check it there, and then nullability will be tracked.
